types/collections.go

```diff
@@ -36,12 +35,6 @@ var (
 	// UintValue represents a collections.ValueCodec to work with Uint.
 	UintValue collcodec.ValueCodec[math.Uint] = uintValueCodec{}
 
-	// TimeKey represents a collections.KeyCodec to work with time.Time
-	// Deprecated: exists only for state compatibility reasons, should not
-	// be used for new storage keys using time. Please use the time KeyCodec
-	// provided in the collections package.
-	TimeKey collcodec.KeyCodec[time.Time] = timeKeyCodec{}
```

Comment:
Imho, we shouldn't remove this, we don't need to introduce such breaking change given that it is already deprecated.